### ISSUE TITLE
Old 3ds connection issues

### ISSUE THREAD

**Issue by drgoku282** [drgoku282/PKMN-NTR#114]
This will be a issue thread which will discuss anything related to PKMN-NTR usage in old 3DS.

Sadly my 2DS screen died, so I can't test this application on an old 3DS.

**Comment by JourneyOver** [comment]
1. Dunno if this actually pertains to only O3DS or not exactly but I can't inject any pokemon at all. I've tried dragging and dropping onto every single tab in pkmn ntr doesn't seem to matter what type of file it is be it a Decrypted single pokémon files, Encrypted single pokémon files, or Mystery gift files.

2. Rosalina inputdirection doesn't seem to work on sun/moon/x/y/or/as, using InputProc works as expected though again idk if this only pertains to O3DS or not as I only have an O3DS.

**Comment by TheEzeJC** [comment]
Is the O3DS even compatible?

**Comment by JourneyOver** [comment]
@TheEzeJC Yes this program is compatible with X/Y/OR/AS/Sun/Moon on the O3(2)DS you just have to use Mode 3 NTR instead of normal NTR if you plan on using this for Sun/Moon and same deal when it gets updated to support USUM.

**Comment by TheEzeJC** [comment]
@JourneyOver Thanks. I’ve been looking everywhere to determine if the software is compatible or not with the O3DS.

**Comment by drgoku282** [comment]
Just to be clear:

- Mode 3 NTR is mandatory for Sun and Moon, on O3DS (same for Ultra Sun and Ultra moon in a future update).
- Dragging pk files onto the tabs actually does nothing unless you click the write buttons on the Write pokemon or the Clone button, let me know if you still have issues.
- This game is not compatible with Rosalina in any way (input redirection, debugging, etc.). You must use the program as described in the wiki.
- If you have a cheat plugin enabled, the program might not work.
- This application wasn't designed to work alongside streaming or other 3DS background applications. So there might be issues.
e

**Comment by JourneyOver** [comment]
> Dragging pk files onto the tabs actually does nothing unless you click the on the Write pokemon button or the Clone button, let me know if you still have issues.

@drgoku282 I can't even drag pk files into PKMN NTR that's what I'm saying, I know how write/clone pokemon, it's just the initial dragging and dropping onto the actual program that is the issue. When I drag the .pk files onto PKMN NTR I just end up with a "🛇" on any tab without the ability to actually drop it in to the program and it doesn't seem to matter if I use the write/clone/read any.

**Comment by drgoku282** [comment]
I can't reproduce the drag and drop issue, are you sure you're dragging it on the PKHeX tabs (right side, left side will not work). Also try to drag on the pokemon picture at the top left.

About Rosalina, support, it is not planned because, honestly, I don't know where to start or if anybody had managed to do RAM manipluation from an external application.

**Comment by JourneyOver** [comment]
@drgoku282 Yea I'm dragging it onto the PKHeX tabs, I've tried dragging onto the pokemon picture as well  and both produce the same result of throwing a "🛇" and not letting me drop it. Give me a bit and I'll see if I can whip up a video or something to show you exactly what is going on.

**Comment by drgoku282** [comment]
I don't know what is happening, but for some reason your computer doesn't seem to be sending the read command, or your 3DS can't send the data back. Try to change your NTR version and tell me what happens.

**Comment by JourneyOver** [comment]
@drgoku282 It sends the read/write commands just fine to and from my 3ds, as I have changed data on several pokemon/items and such through PKMN NTR without any issues, should of caught several pokemon (as I was testing it on my sun copy and I have all my mons on my US game) during that test to show that it was reading correctly. It's just the dragging and dropping of pk and wondercard files into PKMN NTR that it doesn't allow me to do.

As noted before I've tried both 3.4 and 3.5 and both have the same issue, I've even tested the 3.6 Alpha1 version on my US and UM and it's still got the same issue.

**Comment by drgoku282** [comment]
Does it work before connecting to the game? I cannot reproduce the issue.

**Comment by JourneyOver** [comment]
nope, doesn't work even before connecting to the game either. continues to just throw the "🛇" symbol.

I do have one small detail I can add just now though, if I click on the marshadow picture of the (the default mon that comes up when you first load up 3.5 of pkmn ntr if I drag it over any of the tabs it doesn't throw the whole "🛇" symbol same thing with the Zeraora from 3.6alpha1, I can even drag it out of pkmn ntr to my explorer window and drop it there and it'll create the pk7 file. it only does the whole "🛇" symbol trying to drag and drop into pkmn ntr.

**Comment by drgoku282** [comment]
Can you do a gif/video? Try the following

- Open program
- Export pokemon from the sprite to a pk file
- Change species to anything else (just to tell if it is imported correctly).
- Import the same poke

**Comment by drgoku282** [comment]
Also, try to drag and drop from PKHEX directly to PKMN-NTR.

**Comment by JourneyOver** [comment]
do you want me to connect to a game, or just do everything on initial default load of the program?

**Comment by drgoku282** [comment]
Do not connect the game, drag and drop does not depend on the connection.

**Comment by drgoku282** [comment]
This is how the program behaves in my computer:
https://imgur.com/FjeSaHX

**Comment by JourneyOver** [comment]
Bah so I tried to get a gif of it all, but I keep getting errors each time it's converting in shareX.. Give me a few minutes and I'll get a normal video but I do have some stuff.

so I just loaded up pkhex and I noticed I have the same issue on there as I do pkmn ntr, it doesn't allow me to drag/drop pk7 into the program either. 
I forgot to do from PKMN NTR to PKHeX directly when I was working on the gif but I tested it after and it works, PKHeX to PKMN NTR directly is working also.

So the only thing that doesn't seem to work so far is from PK7 file to PKMN NTR.

**Comment by drgoku282** [comment]
So, is PK7 to PKHeX is working then? If it doesn't maybe seeking help in ProjectPokémon might help you because both programs share the same code to handle drag and drop.

**Comment by JourneyOver** [comment]
Nope pk7 to pkhex isn't working. so guess I'll go issue a report over there as well and see if I can get some more help.

**Comment by JourneyOver** [comment]
@drgoku282 Ummmm sooo weird thing, now that I complained about PKHeX not working on dragging/dropping on projectpokemon forums...guess what now works all of a sudden..

**Comment by Gumzone** [comment]
Can someone help me PKMN NTR is not connecting to my 2ds normal and yeah i m introducing the right IP. Is there any thing you need to install beaside ntr mode 3 and input redirection mode 3. I want to use for Ultra moon
